packages/grpc-js/test/test-retry.ts

```diff
@@ -323,6 +323,21 @@ describe('Retries', () => {
         }
       );
     });
+
+    it('Should not retry on custom error code', done => {
+      const metadata = new grpc.Metadata();
+      metadata.set('succeed-on-retry-attempt', '2');
+      metadata.set('respond-with-status', '300');
+      client.echo(
+        { value: 'test value', value2: 3 },
+        metadata,
+        (error: grpc.ServiceError, response: any) => {
+          assert(error);
+          assert.strictEqual(error.details, 'Failed on retry 0');
```

Comment:
This test should also assert that the error code that the application sees is equal to 300. Otherwise, this test could pass if the server doesn't send the custom error code at all, avoiding the relevant code path entirely.

Reply:
ty, fixed